lib/src/node-sass/render.ts

```diff
@@ -73,43 +83,46 @@ export interface RenderError extends Error {
  * This attempts to match the Node Sass `render()` API as closely as possible
  * (see: https://github.com/sass/node-sass#usage).
  */
-export async function render(
+export function render(
   options: RenderOptions,
   callback: (error?: RenderError, result?: RenderResult) => void
-): Promise<void> {
-  const file = options.file ? p.resolve(options.file) : undefined;
-  if (file === undefined && options.data === undefined) {
+): void {
+  const fileRequest = options as FileOptions & SharedOptions;
+  const stringRequest = options as StringOptions & SharedOptions;
+
+  if (!fileRequest.file && isNullOrUndefined(stringRequest.data)) {
     callback(
       newRenderError(Error('Either options.data or options.file must be set.'))
     );
     return;
   }
 
-  let sourceMap;
+  let sourceMap: RawSourceMap | undefined;
   const getSourceMap = wasSourceMapRequested(options)
     ? (map: RawSourceMap) => (sourceMap = map)
     : undefined;
 
-  try {
-    const start = Date.now();
-    const css = options.data
-      ? await compileString({
-          source: options.data,
-          sourceMap: getSourceMap,
-          url: isEmpty(options.file) ? 'stdin' : pathToFileURL(options.file!),
-        })
-      : await compile({path: file!, sourceMap: getSourceMap});
-    callback(undefined, newRenderResult(options, start, css, sourceMap));
-  } catch (error) {
-    callback(newRenderError(error));
-  }
+  const start = Date.now();
+  const compileSass = stringRequest.data
+    ? compileString({
+        source: stringRequest.data,
+        sourceMap: getSourceMap,
+        url: stringRequest.file ? pathToFileURL(stringRequest.file) : 'stdin',
+      })
+    : compile({path: fileRequest.file, sourceMap: getSourceMap});
+
+  compileSass
+    .then(css =>
+      callback(undefined, newRenderResult(options, start, css, sourceMap))
+    )
+    .catch(error => callback(newRenderError(error)));
```

Comment:
Rather than calling `.then()` followed by `.catch()` here, it's better to call `.then()` with both an `onFulfilled` and an `onRejected` callback. Otherwise, if `callback` throws an error synchronously in `then()`, that error will get passed back to `callback`!